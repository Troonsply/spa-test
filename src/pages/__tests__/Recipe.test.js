import { screen, waitForElementToBeRemoved } from '@testing-library/react';
import * as api from '../../api';
import { Recipe } from '../Recipe';
import { renderWithRouter } from '../../utils/testing';

const apiSpy = jest.spyOn(api, 'getMealById');

describe('Recipe', () => {
  it('should render correctly', async () => {
    apiSpy.mockResolvedValueOnce({
      meals: [{
        idMeal: '1',
        strMeal: 'First',
        strCategory: 'Category',
        strArea: 'Area',
        strInstructions: 'Instructions',
        strMealThumb: 'strMealThumb',
        strIngredient1: 'ingredient1',
        strMeasure1: 'measure1',
        strYouTube: 'youtube'
      }]
    });
    renderWithRouter(<Recipe />);
    const preloader = screen.getByRole('progressbar');
    expect(preloader).toBeInTheDocument();
    await waitForElementToBeRemoved(preloader);
    expect(screen.getByRole('article')).toMatchSnapshot();
  });
  it('should render a recipe witout area and youtube', async () => {
    apiSpy.mockResolvedValueOnce({
      meals: [{
        idMeal: '1',
        strMeal: 'First',
        strCategory: 'Category',
        strInstructions: 'Instructions',
        strMealThumb: 'strMealThumb',
        strIngredient1: 'ingredient1',
        strMeasure1: 'measure1',
      }]
    });
    renderWithRouter(<Recipe />);
    const preloader = screen.getByRole('progressbar');
    expect(preloader).toBeInTheDocument();
    await waitForElementToBeRemoved(preloader);
    expect(screen.getByRole('article')).toMatchSnapshot();
  });
})