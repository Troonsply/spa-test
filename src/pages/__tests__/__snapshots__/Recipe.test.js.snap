// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe should render a recipe witout area and youtube 1`] = `
<div
  class="recipe"
  role="article"
>
  <img
    alt="First"
    src="strMealThumb"
  />
  <h1>
    First
  </h1>
  <h6>
    Category: 
    Category
  </h6>
  <p>
    Instructions
  </p>
  <table
    class="centered"
  >
    <thead>
      <tr>
        <th>
          Ingredient
        </th>
        <th>
          Measure
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          ingredient1
        </td>
        <td>
          measure1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Recipe should render correctly 1`] = `
<div
  class="recipe"
  role="article"
>
  <img
    alt="First"
    src="strMealThumb"
  />
  <h1>
    First
  </h1>
  <h6>
    Category: 
    Category
  </h6>
  <h6>
    Area: 
    Area
  </h6>
  <p>
    Instructions
  </p>
  <table
    class="centered"
  >
    <thead>
      <tr>
        <th>
          Ingredient
        </th>
        <th>
          Measure
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          ingredient1
        </td>
        <td>
          measure1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
